### ISSUE TITLE
Error kinesis asyn client kinesis:software.amazon.awssdk.core.exception.SdkClientException: The channel was closed.

### ISSUE THREAD

**Issue by investr777** [aws/aws-sdk-java-v2#2253]
There are many errors while PutRecords: `kinesis:software.amazon.awssdk.core.exception.SdkClientException: The channel was closed. This may have been done by the client (e.g. because the request was aborted), by the service (e.g. because the request took too long or the client tried to write on a read-only socket), or by an intermediary party (e.g. because the channel was idle for too long).`
SDK: software.amazon.awssdk:kinesis:2.10.41
The kinesis client is created like `KinesisAsyncClient.builder().build()`

**Comment by debora-ito** [comment]
Hi @investr777 apologies for the delayed response.

Can you provide a full reproducible sample code and the full stacktrace with the error?

**Comment by investr777** [comment]
Hi, @debora-ito 
```
kinesisAsyncClient.putRecords(PutRecordsRequest.builder()
            .streamName(streamName)
            .records(
                events.map { event ->
                    val eventBytesBuffer = jacksonMapper.writeValueAsBytes(event).let { eventBytes ->
                        ByteBuffer.allocate(eventBytes.size + 2).put(eventBytes).putChar('\n')
                    }
                    val eventSdkBytes = SdkBytes.fromByteArray(eventBytesBuffer.array())
                    PutRecordsRequestEntry.builder()
                        .data(eventSdkBytes)
                        .partitionKey(UUID.randomUUID().toString()).build()
                }
            ).build()
        )
            .thenAccept {
                if (it.failedRecordCount() > 0) {
                    log.error { "${it.failedRecordCount()} records was failed send to kinesis" }
                } else {
                    log.debug { "Successfully sent ${it.records().size} records to kinesis" }
                }
            }
            .exceptionally {
                log.error { "Error:" + it.message }
                null
            }
```
Unfortunately, I can't write the full stacktrace with the error. Because output only a message from an exception

**Comment by debora-ito** [comment]
Hi @investr777 apologies for loosing track of this issue.

Recently we made some changes in how the connection pool is handled in the async clients. The change was made available in the SDK version `2.17.101`. If you are still experiencing this issue please try the latest version (or at least a version >= `2.17.101`) and let us know if you still see the error.
